src/plugins/discord/models.js

```diff
@@ -102,6 +101,17 @@ export function emojiToRef({id, name}: Emoji): EmojiRef {
   return `${name}:${id}`;
 }
 
+/**
+ * Returns an Emoji object based on a string reference in the form:
+ * `${name}:${id}`
+ */
+export function refToEmoji(ref: string): Emoji {
+  // TODO: Test that ref is in correct form
+  const [name, id] = ref.split(":");
```

Comment:
I was thinking of testing that the colon was present; seeing as generic emoji will not have a colon though, because their `id` prop would be `nil`, I'm not sure what there is to test here.

Reply:
In reality I think the refs have more specific rules than just the colon. I'm guessing a built-in is a single unicode character (emoji). Where a custom is probably ascii (don't know exact allowed character set) a colon and a snowflake.

Refs are sometimes user input we should ideally validate:
- From project.json / future configuration file (best validated)
- From the DB (reasonable to make assumptions)

So I think a stricter test on character sets would be good to have.
That said, I would be ok with doing this in a follow-up PR.